pkg/controller/cheinstallation/cheinstallation_controller.go

```diff
@@ -72,8 +73,11 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	if err := c.Watch(&source.Kind{Type: &orgv1.CheCluster{}}, enqueueRequestForOwner); err != nil {
-		return err
+	r.watchCheCluster = func() error {
```

Comment:
I like the idea of using a func to keep a reference to the `c` controller and use it later in the reconcile loop 👍